std/internal/test/dummyrange.d

```diff
@@ -28,23 +28,41 @@ enum ReturnBy
     Value
 }
 
+import std.traits : isArray;
+
 // Range that's useful for testing other higher order ranges,
 // can be parametrized with attributes.  It just dumbs down an array of
 // numbers 1..10.
-struct DummyRange(ReturnBy _r, Length _l, RangeType _rt)
+struct DummyRange(ReturnBy _r, Length _l, RangeType _rt, T = uint[])
+        if (isArray!T)
 {
+    private enum uinttestData =
+        [1U, 2U, 3U, 4U, 5U, 6U, 7U, 8U, 9U, 10U];
     // These enums are so that the template params are visible outside
     // this instantiation.
     enum r = _r;
     enum l = _l;
     enum rt = _rt;
 
-    uint[] arr = [1U, 2U, 3U, 4U, 5U, 6U, 7U, 8U, 9U, 10U];
+    static if (is(T == uint[]))
+    {
+        T arr = uinttestData;
+    }
+    else
+    {
+        T arr;
+    }
+
+    alias RetType = ElementType!(T);
+    alias RetTypeNoAutoDecoding = ElementEncodingType!(T);
 
     void reinit()
     {
         // Workaround for DMD bug 4378
-        arr = [1U, 2U, 3U, 4U, 5U, 6U, 7U, 8U, 9U, 10U];
+        static if (is(T == uint[]))
```

Comment:
The behavior is not different, but the original DummyRange was an uint[] only (1 to 11) range, and I wanted to preserve that bahavior so the DummyRange api would not break. But reinit broke every DummyRange(T) where a uint[] could not implicitliy be assigned to a T[]